pkgs/os-specific/windows/cygwin/headers.nix

```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  fetchpatch,
+  mingw_w64_headers,
+}:
+
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "cygwin-headers";
+  version = "3.5.4";
+
+  src = fetchFromGitHub {
+    owner = "mirror";
+    repo = "newlib-cygwin";
+    rev = "cygwin-${finalAttrs.version}";
+    hash = "sha256-ZfT6JhOXLCJOY0vSVz6aKShmtuTN9/0NZ1k1RMSZX4Q=";
+  };
+
+  patches = [
+    (fetchpatch {
```

Comment:
This is a static file, so there's no need for patch normalization and fetchurl can be used.